### ISSUE TITLE
Support `bob use <full-commit-hash>`

### ISSUE THREAD

**Issue by nullchilly** [MordechaiHadad/bob#138]
Currently bob works with short commit hash:

```sh
~
❯ bob use 19eef81
Jun 20 19:41:56.368  INFO 19eef81 is already installed and used!

~
❯ nvim --version
NVIM v0.10.0-dev-542+g19eef8156
```

However, bob failed to find the binary with full commit hash:

```sh
~
❯ bob use 19eef8156bdd3539f10c9a39494cb64b88d777f5
Jun 20 19:42:18.190  INFO You can now use 19eef8156bdd3539f10c9a39494cb64b88d777f5!

~
❯ nvim --version
thread 'main' panicked at 'Failed to spawn child process: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/main.rs:87:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

![image](https://github.com/MordechaiHadad/bob/assets/56817415/cde7fe86-c288-4391-8055-676f58bfacaf)

I often use `Copy the full SHA` which in this case returns `19eef8156bdd3539f10c9a39494cb64b88d777f5`, I think that the full commit hash is more convenient to use.

**Comment by MordechaiHadad** [comment]
Bob supports the full commit and cuts it, but for some reason it seems like when specifying full hash it gets fucked... I will have to fix this

**Comment by MordechaiHadad** [comment]
fixed now with new PR

**Comment by nullchilly** [comment]
Thanks, it works now 🎉 

```sh
bob on  fix/full-hash is 📦 v2.4.0 via 🦀 v1.69.0
❯ ~/.cargo/bin/bob use 19eef8156bdd3539f10c9a39494cb64b88d777f5
Jun 21 22:51:25.743  INFO 19eef81 is already installed and used!

bob on  fix/full-hash is 📦 v2.4.0 via 🦀 v1.69.0
❯ nvim --version
NVIM v0.10.0-dev-542+g19eef8156
```
